app/mailers/discovered_mailer.rb

```diff
@@ -1,7 +1,18 @@
 class DiscoveredMailer < ::ApplicationMailer
   helper :discovered_hosts
   def discovered_summary(options = {})
-    raise ::Foreman::Exception.new(N_("Must specify a user with email enabled")) unless (user = User.find(options[:user])) && user.mail_enabled?
+    user = if options[:user].kind_of? User
+             options[:user]
+           elsif options[:user].kind_of? Integer
+             User.find(options[:user])
+           else
+             raise ::Foreman::Exception.new(N_('Invalid user type of %s was provided') % options[:user].class.to_s)
+           end
+
+    unless user.mail_enabled?
+      log.debug(N_("The user %d does not enable email") % user.id)
```

Comment:
Wouldn't this raise MethodNotFound: log error? Can you add a test for this code branch?

Reply:
it should be `Rails.logger` you are correct :(